### PR TITLE
Added vue documentation

### PR DESCRIPTION
**Pull request by armanio123**
Added initial documentation for the Vue.js support

### CODE REVIEW

**Comment by armanio123** [comment]
Mentioning @billti for review

File: articles/frameworks/vue.md [hunk 1]
```diff
@@ -1,3 +1,104 @@
...
+## Feature overview
+
+In addition to the normal Visual Studio features when writing TypeScript or JavaScript, you'll see new behavior for Vue.js specific features:
+
+* Supported Script, Style and Template blocks in .vue files.
+* Recognize lang attribute on .vue files.
+
+## Create a Vue.js project with the vue-cli
+
+Vue.js provides and official CLI for quickly scaffolding projects. If you'll like to use it to create your application, follow the steps below to setup your developer environment.
```
Comment (billti) [inline articles/frameworks/vue.md]: "__an__ official CLI", and "if __you would__ like to". (And "setup" should "set up").

File: articles/frameworks/vue.md [hunk 2]
```diff
@@ -1,3 +1,104 @@
...
+// Template for index.html
+index: path.resolve(__dirname, '../../wwwroot/index.html'),
+
+// Paths
+assetsRoot: path.resolve(__dirname, '../../wwwroot'),
+```
+
+##### Indicate the project to build the ClientApp everytime building is triggered:
+1. Go to **Project properties > Build Events**
+1. On **Pre-build event command line** type `npm --prefix ./ClientApp run build`.
```
Comment (billti) [inline articles/frameworks/vue.md:57]: What does the `--prefix` option do?

Comment (billti) [reply articles/frameworks/vue.md:57]: Ah, OK. https://stackoverflow.com/a/14469516/1674945

Comment (billti) [reply articles/frameworks/vue.md:57]: (CC @uniqueiniquity who was asking about this earlier :smile:)

Comment (uniqueiniquity) [reply articles/frameworks/vue.md:57]: @billti I was trying that, but I don't think it behaves exactly as you would expect it to when used with `npm install`. That's a discussion for a different time and place though :)

File: articles/frameworks/vue.md [hunk 3]
```diff
@@ -1,3 +1,104 @@
...
+	}
+};
+```
+
+##### Building with vue-cli 3.0
+There is an unknown issue with the vue-cli 3.0 that prevents automating the build process. Everytime you wish to refresh the wwwroot folder you will need to run the command `npm run build` on the ClientApp folder.
+
+## Limitations
+
+* Lang attribute only supports JavaScript and TypeScript languages.
```
Comment (billti) [inline articles/frameworks/vue.md]: Is it worth documenting here how it should look? (e.g. Can you use "ts" (`<script lang="ts">`)? Or does it need to be "typescript"?).

File: articles/frameworks/vue.md [hunk 4]
```diff
@@ -1,3 +1,104 @@
...
+* Lang attribute doesn't work with template or style tags.
+* Debugging .vue files is not supported due to its preprocessed nature.
```
Comment (billti) [inline articles/frameworks/vue.md]: "Debugging script blocks in .vue files..."

File: articles/frameworks/vue.md [hunk 5]
```diff
@@ -1,3 +1,104 @@
...
+* .vue files are not recognized as modules. Shimming \*.vue files is necessary (vue-cli 3.0 template already includes this file).
+* Running the command `npm run build` as a precompile task doesn't work when using vue-cli 3.0, for an unknown reason.
```
Comment (billti) [inline articles/frameworks/vue.md]: The two bullets up here I don't understand. Can you expand on what "shimming" is, and drop the "for an unknown reason" (just say it doesn't work).

File: articles/frameworks/vue.md [hunk 6]
```diff
@@ -1,3 +1,104 @@
...
+
+## Additional resources
+* https://vuejs.org/v2/guide - Vue get started guide.
+* https://github.com/vuejs/vue-cli - Vue CLI project.
+* https://webpack.js.org/configuration/ - Webpack configuration documentation.
+* https://github.com/madskristensen/vuepack2017 - Popular Vue.js extension for Visual Studio
```
Comment (billti) [inline articles/frameworks/vue.md]: Should we link to this? Won't it conflict if they install this too and it tries to register the .vue extension? Or are we a higher priority?

**Review by billti — CHANGES_REQUESTED** [review]
A few minor comments. Thanks for writing up!

File: articles/frameworks/vue.md (hunk 1 above)
Comment (armanio123) [reply articles/frameworks/vue.md]: Changed.

File: articles/frameworks/vue.md (hunk 3 above)
Comment (armanio123) [reply articles/frameworks/vue.md]: I think it has some value. Added.

File: articles/frameworks/vue.md (hunk 4 above)
Comment (armanio123) [reply articles/frameworks/vue.md]: Fixed.

File: articles/frameworks/vue.md (hunk 5 above)
Comment (armanio123) [reply articles/frameworks/vue.md]: Done.

File: articles/frameworks/vue.md (hunk 6 above)
Comment (armanio123) [reply articles/frameworks/vue.md]: That's a good question, seems like right now is creating a conflict, I'll removed it. Anyway, the extension adds some nice features to Vue so we'll probably want to take a look later to remove the conflict.

**Pull request merged**